src/hardware/pounder/hrtimer.rs

```diff
@@ -54,37 +54,32 @@ impl HighResTimerE {
     pub fn configure_single_shot(
         &mut self,
         channel: Channel,
-        set_duration: f32,
-        set_offset: f32,
+        delay: f32,
+        duration: f32,
     ) {
         // Disable the timer before configuration.
         self.master.mcr.modify(|_, w| w.tecen().clear_bit());
 
         // Configure the desired timer for single shot mode with set and reset of the specified
         // channel at the desired durations. The HRTIM is on APB2 (D2 domain), and the kernel clock
         // is the APB bus clock.
-        let minimum_duration = set_duration + set_offset;
-
-        let source_frequency: u32 = self.clocks.timy_ker_ck().to_Hz();
-        let source_cycles =
-            (minimum_duration * source_frequency as f32) as u32 + 1;
+        let clk = self.clocks.timy_ker_ck().to_Hz() as f32;
+        let end = ((delay + duration) * clk) as u32 + 1;
 
         // Determine the clock divider, which may be 1, 2, or 4. We will choose a clock divider that
         // allows us the highest resolution per tick, so lower dividers are favored.
-        let setting: u8 = if source_cycles < 0xFFDF {
+        let div: u8 = if end < 0xFFDF {
             1
-        } else if (source_cycles / 2) < 0xFFDF {
+        } else if (end / 2) < 0xFFDF {
             2
-        } else if (source_cycles / 4) < 0xFFDF {
+        } else if (end / 4) < 0xFFDF {
             3
         } else {
             panic!("Unattainable timing parameters!");
         };
 
-        let divider = 1 << (setting - 1);
-
         // The period register must be greater than or equal to 3 cycles.
-        let period = (source_cycles / divider as u32) as u16;
+        let period = (end / (1 << (div - 1)) as u32) as u16;
```

Comment:
Hmm... This indeed looks like it didn't work right for div = 3. Looks like we were unintentionally using a slightly shorter period in some cases. Nice fix!

Reply:
What do you mean. The code should be equivalent. This wasn't intended to be a fix ;)

Reply:
Ah, sorry, I had completely missed the deletion of the `let divider = 1 << (setting - 1)` above and thought we were using `divider = setting` in the previous version. Indeed, there is no actual change here.